src/ddbToEs/ddbToEsHelper.test.ts

```diff
@@ -88,4 +90,43 @@ describe('DdbToEsHelper', () => {
             );
         });
     });
+
+    describe('isRemoveResource', () => {
+        const record: any = {
+            eventID: 'some-event-id',
+            eventName: 'INSERT',
+            dynamodb: {
+                OldImage: { documentStatus: { S: 'AVAILABLE' } },
+                NewImage: { documentStatus: { S: 'AVAILABLE' } },
+            },
+        };
+
+        test('Should remove for REMOVE event', () => {
+            const removeRecord: any = cloneDeep(record);
+            removeRecord.eventName = 'REMOVE';
+            expect(ddbToEsHelper.isRemoveResource(removeRecord)).toBeTruthy();
+        });
+
+        test('Should remove for new image in DELETED status and hard delete enabled', () => {
+            process.env.ENABLE_ES_HARD_DELETE = 'true';
+            const modifyRecord: any = cloneDeep(record);
+            modifyRecord.eventName = 'MODIFY';
+            modifyRecord.dynamodb = {
+                OldImage: { documentStatus: { S: 'AVAILABLE' } },
+                NewImage: { documentStatus: { S: 'DELETED' } },
+            };
```

Comment:
IMO it's simpler to just declare the entire `modifyRecord`. Cloning and then updating gives little benefit since you need to update almost the entire record anyways.

```suggestion
            const modifyRecord: any = {
              eventID: 'some-event-id',
              eventName: 'MODIFY',
              dynamodb: {
                  OldImage: { documentStatus: { S: 'AVAILABLE' } },
                  NewImage: { documentStatus: { S: 'DELETED' } },
              },
            }
```